File: notes_and_examples/find_number_links.py
from selenium import webdriver # pip install selenium

# Instantiate the Chrome WebDriver
driver = webdriver.Chrome()

# Open Google
driver.get('https://www.google.com')

# Find all the anchor elements (links)
links_found = driver.find_elements_by_tag_name('a')

# Get the count of the links
num_of_links = len(links_found)
print("Number of links on Google homepage:", num_of_links)

# Close the browser
driver.quit()

"""

This code navigates to the Google homepage, finds all the anchor elements (links) using driver.find_elements_by_tag_name('a'), 
counts the number of links, and prints the total number of links found on the Google homepage.

"""

File: notes_and_examples/find_number_links.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

public class CountLinksOnGoogle {
    public static void main(String[] args) {

        // Set the path to your ChromeDriver executable
        System.setProperty("webdriver.chrome.driver", "path_to_chromedriver");

        // Initialize the Chrome WebDriver
        WebDriver driver = new ChromeDriver();

        // Open Google
        driver.get("https://www.google.com");

        // Find all the anchor elements (links)
        // You can modify this if needed to find elements in a different way
        java.util.List<WebElement> allLinks = driver.findElements(By.tagName("a"));

        // Get the count of the links
        int numberOfLinks = allLinks.size();
        System.out.println("Number of links on the Google homepage: " + numberOfLinks);

        // Close the browser
        driver.quit();
    }
}
